File: src/minigames/maniac_digger/java/work/lclpnet/ap2/game/maniac_digger/ManiacDiggerMiniGame.java
package work.lclpnet.ap2.game.maniac_digger;

import net.minecraft.core.RegistryAccess;
import net.minecraft.resources.Identifier;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.Items;
import org.jetbrains.annotations.NotNull;
import work.lclpnet.ap2.ApConstants;
import work.lclpnet.ap2.api.game.*;

public class ManiacDiggerMiniGame implements MiniGame {

    @Override
    public boolean canBeFinale(@NotNull GameStartContext context) {
        return true;
    }

    @Override
    public boolean canBePlayed(@NotNull GameStartContext context) {
        return context.getParticipantCount() <= 12;  // maps should support 12 pipes
    }

    @Override
    public @NotNull MiniGameInstance createInstance(@NotNull MiniGameHandle gameHandle) {
        return new ManiacDiggerInstance(gameHandle);
    }

    @Override
    public @NotNull Identifier getId() {
        return ApConstants.identifier("maniac_digger");
    }

    @Override
    public @NotNull GameType getType() {
        return GameType.FFA;
    }

    @Override
    public @NotNull String getAuthor() {
        return ApConstants.PERSON_LCLP;
    }

    @Override
    public @NotNull ItemStack getIcon(@NotNull RegistryAccess manager) {
        return new ItemStack(Items.GOLDEN_SHOVEL);
    }
}

File: src/minigames/maniac_digger/src/main/kotlin/work/lclpnet/ap2/game/maniac_digger/ManiacDiggerMiniGame.kt
package work.lclpnet.ap2.game.maniac_digger

import net.minecraft.core.RegistryAccess
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.Items
import work.lclpnet.ap2.ApConstants
import work.lclpnet.ap2.api.game.GameStartContext
import work.lclpnet.ap2.api.game.GameType
import work.lclpnet.ap2.game.MiniGame

class ManiacDiggerMiniGame : MiniGame {
    override fun canBeFinale(context: GameStartContext) = true
    override fun canBePlayed(context: GameStartContext) = context.participantCount <= 12
    override fun createFactory() = ManiacDiggerFactory()
    override val id = ApConstants.identifier("maniac_digger")
    override val type = GameType.FFA
    override val author = ApConstants.PERSON_LCLP
    override fun getIcon(manager: RegistryAccess) = ItemStack(Items.GOLDEN_SHOVEL)
}
